src/main/java/net/snowflake/ingest/SimpleIngestManager.java

```diff
@@ -232,30 +240,116 @@ public SimpleIngestManager(String account, String user, String pipe,
+   * @param keyPair    the KeyPair we'll use to sign JWT tokens
    * @param schemeName http or https
-   * @param hostName the hostname
-   * @param port the port number
+   * @param hostName   the hostname
+   * @param port       the port number
    */
+  @Deprecated
   public SimpleIngestManager(String account, String user, String pipe,
-                      KeyPair keyPair, String schemeName,
-                      String hostName, int port)
+                             KeyPair keyPair, String schemeName,
+                             String hostName, int port)
   {
     //call our initializer method
     init(account, user, pipe, keyPair);
 
     //make the request builder we'll use to build messages to the service
     builder = new RequestBuilder(account, user, keyPair,
-                                 schemeName, hostName, port);
+        schemeName, hostName, port);
+  }
+
+  /**
+   * Constructs a SimpleIngestManager for a given user in a specific account
+   * In addition, this also takes takes the target table and source stage
+   * Finally, it also requires a valid private key registered with
+   * Snowflake DB
+   *
+   * @param account    the account into which we're loading
+   * @param user       the user performing this load
+   * @param pipe       the fully qualified name of the pipe
+   * @param privateKey the private key we'll use to sign JWT tokens
+   * @param schemeName http or https
+   * @param hostName   the hostname
+   * @param port       the port number
+   * @throws NoSuchAlgorithmException if can't create key factory by using
+   *                                  RSA algorithm
+   * @throws InvalidKeySpecException  if private key or public key is invalid
+   */
+  public SimpleIngestManager(String account, String user, String pipe,
+                             PrivateKey privateKey, String schemeName,
+                             String hostName, int port)
+      throws NoSuchAlgorithmException, InvalidKeySpecException
+  {
+    KeyPair keyPair = createKeyPairFromPrivateKey(privateKey);
+    //call our initializer method
+    init(account, user, pipe, keyPair);
+
+    //make the request builder we'll use to build messages to the service
+    builder = new RequestBuilder(account, user, keyPair,
+        schemeName, hostName, port);
+
+  }
+
+  /**
+   * generate key pair object from private key
+   *
+   * @param privateKey private key
+   * @return a key pair object
+   * @throws NoSuchAlgorithmException if can't create key factory by using
+   *                                  RSA algorithm
+   * @throws InvalidKeySpecException  if private key or public key is invalid
+   */
+  private KeyPair createKeyPairFromPrivateKey(PrivateKey privateKey) throws
+      NoSuchAlgorithmException, InvalidKeySpecException
+  {
+    KeyFactory kf = KeyFactory.getInstance("RSA");
+
+    //generate public key from private key
+    RSAPrivateCrtKey privk = (RSAPrivateCrtKey) privateKey;
```

Comment:
Throw proper user error in case user is passing in a PrivateKey not instancesof RSAPrivateCrtKey